src/Neo.VM/JumpTable/JumpTable.Compound.cs

```diff
@@ -551,7 +571,9 @@ public virtual void PopItem(ExecutionEngine engine, Instruction instruction)
         {
             var x = engine.Pop<VMArray>();
             var index = x.Count - 1;
-            engine.Push(x[index]);
+            var item = x[index];
+            engine.Push(item);
+            engine.ReferenceCounter.RemoveStackReference(item);
```

Comment:
I've stopped at this line for a while. And I think it's not correct, see https://github.com/nspcc-dev/neo-go/pull/3688. Yeah, you've just ported it over, but that seems to be a NeoGo bug.

Reply:
Looks like you are right, it was popped, then pushed to the stack, so should not remove reference here.